From ba8c54a2c68c1676f3b304a06897a3011584746f Mon Sep 17 00:00:00 2001
From: Takanari Hayama <taki@igel.co.jp>
Date: Mon, 17 Nov 2014 08:59:56 +0900
Subject: [PATCH 22/22] v4l2-renderer: skip view not attached to the renderer.

The compositor may pass views that are not previously attached
to the renderer. We should ignore such views.
---
 src/v4l2-renderer.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/v4l2-renderer.c b/src/v4l2-renderer.c
index 11c8c67..d4c540b 100644
--- a/src/v4l2-renderer.c
+++ b/src/v4l2-renderer.c
@@ -653,6 +653,9 @@ draw_view(struct weston_view *ev, struct weston_output *output,
 	pixman_region32_t surface_blend;
 	pixman_box32_t *region;
 
+	if (vs->planes[0].dmafd == 0)
+		return;
+
 	/*
 	 * Check if the surface is still valid. OpenGL/ES apps may destroy
 	 * buffers before they destroy a surface. This check works in the
-- 
1.7.9.5

